pkg/component/registrycaches/registry_caches_test.go

```diff
@@ -424,3 +615,32 @@ status: {}
 		})
 	})
 })
+
+func encode(obj runtime.Object) []byte {
+	data, _ := json.Marshal(obj)
+	return data
+}
+
+func encodeB64(val string) string {
+	return base64.StdEncoding.EncodeToString([]byte(val))
+}
+
+func getSecretResourceKey(data map[string][]byte, prefix string) string {
+	for key := range data {
+		if strings.HasPrefix(key, prefix) {
+			return key
+		}
+	}
+	return ""
+}
+
+func getSecretNameHash(name string) string {
+	index := strings.LastIndexByte(name, '-')
+	if index != -1 {
+		hash := name[index+1:]
+		if strings.HasSuffix(hash, ".yaml") {
+			return hash[:len(hash)-5]
+		}
+	}
+	return ""
+}
```

Comment:
I see that in gardener/gardener the immutable Secret names are usually hard-coded: ref https://github.com/gardener/gardener/blob/760571e4de9986270a73cf9b2e043d3b84e3d2db/pkg/component/kubeapiserver/kube_apiserver_test.go#L107 and ref https://github.com/gardener/gardener/blob/d3c8887077e59c8af145f49c5258266d952bb03b/pkg/component/kubeproxy/kube_proxy_test.go#L199.

If our secrets in the test has the same data always (hence, the same Secret name always), does it make sense to hard-code it?